ios/Video/RCTVideo.m

```diff
@@ -483,14 +484,16 @@ - (void)playerItemForSource:(NSDictionary *)source withCallback:(void(^)(AVPlaye
     [assetOptions setObject:cookies forKey:AVURLAssetHTTPCookiesKey];
 
 #if __has_include(<react-native-video/RCTVideoCache.h>)
-    if (!_textTracks) {
-      /* The DVURLAsset created by cache doesn't have a tracksWithMediaType property, so trying
-       *  to bring in the text track code will crash. I suspect this is because the asset hasn't fully loaded.
-       * Until this is fixed, we need to bypass caching when text tracks are specified.
-       */
-      DebugLog(@"Caching is not supported for uri '%@' because text tracks are not compatible with the cache. Checkout https://github.com/react-native-community/react-native-video/blob/master/docs/caching.md", uri);
-      [self playerItemForSourceUsingCache:uri assetOptions:assetOptions withCallback:handler];
-      return;
+    if (shouldCache) {
```

Comment:
I would convert this to:
`if (shouldCache && (!_textTracks || !_textTracks.count)) {`
for readability.